elastic/kibana
# [Response Ops][Connectors] Allow connectors to explicitly register which features they will be available in

Resolves https://github.com/elastic/kibana/issues/135467 and https://github.com/elastic/kibana/issues/82502

## Summary

Adds field for connectors to explicitly register which feature they will be available in. This replaces the hard-coded list of connector types that was found in various parts of the UI:

For security solutions, this was defined as
```
NOTIFICATION_SUPPORTED_ACTION_TYPES_IDS = [
  '.email',
  '.index',
  '.jira',
  '.pagerduty',
  '.resilient',
  '.servicenow',
  '.servicenow-sir',
  '.servicenow-itom',
  '.slack',
  '.swimlane',
  '.teams',
  '.webhook',
]
```

For cases, this was defined as
```
SUPPORTED_CONNECTORS = [
  `${ConnectorTypes.serviceNowITSM}`,
  `${ConnectorTypes.serviceNowSIR}`,
  `${ConnectorTypes.jira}`,
  `${ConnectorTypes.resilient}`,
  `${ConnectorTypes.swimlane}`,
]
```

For uptime, this was defined as 
```
ALLOWED_ACTION_TYPES: ActionTypeId[] = [
  '.slack',
  '.pagerduty',
  '.server-log',
  '.index',
  '.teams',
  '.servicenow',
  '.jira',
  '.webhook',
  '.email',
]
```

To replace these, I created 4 feature ids, `alerting`, `cases`, `uptime` and `siem`. Each connector type defines an array of feature ids on registration.

### Connector List
New column showing "Availability" for created connectors.
<img width="2035" alt="Screen Shot 2022-07-14 at 11 41 52 AM" src="https://user-images.githubusercontent.com/13104637/179619932-a3957dc6-40be-48bc-95ff-816489a7a470.png">

### Create Connector Flyout
Added "Availability" information to header
<img width="673" alt="Screen Shot 2022-07-18 at 5 26 59 PM" src="https://user-images.githubusercontent.com/13104637/179620381-3ce2bb44-5850-4862-a05f-69f1c8d9756b.png">

## To Verify
* Verify that creating a rule from Stack Management allows you to create actions using all of the existing connector types. Currently all connector types have `alerting` in their feature config and there are no connector types that are not available in the Stack Rules and Management UI
* Verify that creating a rule from Security Solutions only allows you to create actions using one of the connector types listed above in `NOTIFICATION_SUPPORTED_ACTION_TYPES_IDS`
* Verify that editing the Cases configuration only allows you to add and use connectors that are one of the connector types listed above in `SUPPORTED_CONNECTORS`
* Verify that modifying the Uptime Settings only allows you to add and use connectors that are one of the connector types listed above in `ALLOWED_ACTION_TYPES`


### Checklist
